### ISSUE TITLE
about open source miner

### ISSUE THREAD

**Issue by SpiritedAwayLab** [UnamSanctam/SilentCryptoMiner#1983]
Hi, UnamSanctam. first of all, thank you so much about your work. That is brilliant and it saves my time for a lot. 
I got a question, about the miner you use, actually it is closed source right? I know you didn't put any harmful code in that miner, but is it possible to open source the miner? as I am going to add some small function:  when miner request a remote configuration, it will post the GPU type to server, the server will then generate suitable configuration for this miner and response it back. 
however, I found the configuration request process has been compiled in the miner (ethminer/xmrig) . so do you have any altinative solution for my request?

**Comment by UnamSanctam** [comment]
Yes right now I don't maintain repositories for them, since I sadly wouldn't have enough time to maintain them (due to work), just like I still haven't been able to finish and publish my changes to UnamBinder and UnamDownloader. Though for your purposes it seems like you could just use the "API Endpoint URL", that will post all the information of the miner to the specified URL every minute, including the GPU(s) of the computer.

**Comment by SpiritedAwayLab** [comment]
it seems can be working. I am trying to re-write the endpoint.php by using nodejs, so that I can response the config depends on GPU type.

**Comment by UnamSanctam** [comment]
It should be quite easy, the miner only sends a JSON array, so JSON decode that array and then you can find the GPUs (if there are multiple then they are separated by a comma and space ", ") under the "gpu" value.
The code might be something like this:
```
const http = require('http');

http.createServer((req, res) => {
    if (req.method === 'POST') {
        let body = '';

        req.on('data', chunk => {
            body += chunk.toString();
        });

        req.on('end', () => {
            try {
                const data = JSON.parse(body);
                let response = { response: "ok" };

                if (data.gpu) {
                    const gpus = data.gpu.split(', ');
                    
                    // Do your GPU checking logic here and set the "response" variable to your specific configuration
                }

                res.writeHead(200, { 'Content-Type': 'application/json' });
                res.end(JSON.stringify(response));
            } catch (error) {
                res.writeHead(400, { 'Content-Type': 'application/json' });
                res.end(JSON.stringify({ error: 'Invalid JSON' }));
            }
        });
    } else {
        res.writeHead(405, { 'Content-Type': 'application/json' });
        res.end(JSON.stringify({ error: 'Method not allowed' }));
    }
}).listen(3000);
```

**Comment by SpiritedAwayLab** [comment]
hi
 just wondering, is the config update/ report upload  progress has been integrated into miner program? which is close source one?
